packages/pluggable-widgets-tools/src/typings-generator/generateClientTypes.ts

```diff
@@ -162,6 +172,14 @@ ${generateClientTypeBody(childProperties, isNative, generatedTypes, resolveChild
             return prop.$.isList === "true" ? `${childType}[]` : childType;
         case "widgets":
             return prop.$.dataSource ? "ListWidgetValue" : "ReactNode";
+        case "selection":
+            if (!prop.selectionTypes?.length) {
+                throw new Error("[XML] Selection property requires selectionTypes element");
+            }
+
+            const selectionTypes = prop.selectionTypes.flatMap(s => s.selectionType).map(s => s.$.name);
+            const clientTypes = selectionTypes.filter(s => s !== "None").map(toSelectionClientType);
```

Comment:
For the case where there is only the None option, the array of selection types will be empty and there is nothing for the map function to iterate over.

When there is only the None type, should the client type be "any"?

Reply:
I've amended my commit to return `any` for the union type if the list of types is empty. This will never happen for any valid widget, but it's a reasonable placeholder nonetheless.